packages/frontend/src/react-context-app.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { AppContext } from './app-context';
+import { App } from '@h4bff/core';
+
+const getAppFromContext = (context: any) => {
+  if (!context.app) {
+    throw new Error('App is not defined in this component context.');
+  }
+
+  if (!(context.app instanceof App)) {
```

Comment:
Maybe a comment why we need this check? Is it if another tool accidentally overrides contextType in a way that also provides an `app` property?